File: biweekly-contest/biweekly-contest-19/2-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/prefix-sum.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        int targetSum = threshold * k;
        vector<int> prefix = {0};
        for (int n : arr) {
            prefix.push_back(prefix.back() + n);
        }

        int res = 0;
        for (int i = k; i < prefix.size(); ++i) {
            if (prefix[i] - prefix[i - k] >= targetSum) ++res;
        }
        return res;
    }
};

File: biweekly-contest/biweekly-contest-35/1/prefix-sum.js
/**
 * 暴力/前缀和
 */
var sumOddLengthSubarrays = function(arr) {
  const p = [0];
  for (const n of arr) {
    p.push(p[p.length - 1] + n);
  }
  const n = arr.length;
  let res = 0;
  for (let i = 0; i <= n; ++i) {
    for (let j = i + 1; j <= n; j += 2) {
      res += p[j] - p[i];
    }
  }
  return res;
};